src/Processors/Sources/ShellCommandSource.cpp

```diff
@@ -73,6 +75,13 @@ static int pollWithTimeout(pollfd * pfds, size_t num, size_t timeout_millisecond
     while (true)
     {
         Stopwatch watch;
+
+        auto describe_fd = [](const auto & pollfd) { return fmt::format("(fd={}, flags={})", pollfd.fd, fcntl(pollfd.fd, F_GETFL)); };
```

Comment:
Maybe move describe_fd before this loop, so it could also be deleted from line 110?